core/src/main.cpp

```diff
@@ -511,7 +528,9 @@ auto main(int _argc, char* _argv[]) -> int
 		// JSON configs needs to be in main scope to last the entire duration of the program
 		nlohmann::json oi_config;
 		nlohmann::json endpoint_config;
-		load_oidc_configuration(http_server_config, oi_config, endpoint_config);
+		if (!load_oidc_configuration(http_server_config, oi_config, endpoint_config)) {
```

Comment:
Did we know if we wanted to fail here, or just log that OIDC couldn't be set up?

Reply:
Agree on caller handling anything.

Did we have a way to say don't use OIDC?

Currently, if there's an invalid config, we just keep running and assume that the user wants to run with OIDC disabled.
Might be a new issue if we don't have a way to properly turn it off.

Reply:
my vote....

if they have defined any oidc settings, and they are not good - we should just stop.   no ambiguity.  no surprises for the admin.

they can not have *any*, or they should have *all* and be good/valid.

Reply:
In my PR, I added config validation so that should help.

What is considered an invalid config? Can the application detect that?

Reply:
a fair point... maybe it's not this code's job.

Reply:
especially if we ALWAYS run validation

of course, if they pass in a looser schema, and pass mandatory validation... getting to this code, which doesn't check... could be bad?   hmmm

Reply:
Invalid could be something like an endpoint that doesn't lead to a valid OIDC config, an endpoint that doesn't use a supported scheme... Effectively if there's something wrong with `provider_url` or what it points to, I think that would fall under invalid.

Reply:
> my vote....
> 
> if they have defined any oidc settings, and they are not good - we should just stop. no ambiguity. no surprises for the admin.
> 
> they can not have _any_, or they should have _all_ and be good/valid.

Would that mean if the OIDC section is missing, we skip OIDC, or we have something like an `is_enabled` do the job of that?

Thinking about how this might affect the config validation that Kory made.
That and what to check for in the HTTP code.

Reply:
If OIDC is defined, it is enabled. If OIDC is not defined, ignore OIDC code.

I made the OIDC stanza optional in the validation commit, so we should be good.

Reply:
> If OIDC is defined, it is enabled. If OIDC is not defined, ignore OIDC code.

i think i agree.

at least until we hear that it's hard to hold for admins... then perhaps an 'enabled' flag could make sense.

Reply:
Added logic to skip OIDC configuration if `openid_connect` is not in server config.